src/coco/repository/util/repositoryUtil.ts

```diff
@@ -0,0 +1,86 @@
+import RepositoryType from "../model/RepositoryType";
+import Repository from "../model/Repository";
+import { execSync } from "child_process";
+import Logger from "../../../log/Logger";
+
+const GET_GIT_REPOSITORY_COMMAND = "git config --local --get remote.origin.url";
+const GIT_REPO_URL_SUFFIX = ".git";
+const DOMAIN_GITHUB = "github.com";
+const DOMAIN_GITLAB = "gitlab.com";
+const DOMAIN_BITBUCKET = "bitbucket.org";
+const CUSTOM_DOMAIN_REG_EXP_GITHUB = /github.(.+).com/; // TODO
+const SUFFIX_HTTPS = "https://";
+
+function getRepositoryForType(rootFolderPath: string, type: RepositoryType): Repository {
+    let fullUrl: string | undefined;
+    try {
+        fullUrl = execSync(GET_GIT_REPOSITORY_COMMAND, {
+            cwd: rootFolderPath
+        }).toString().trim();
+    } catch (error) {
+        Logger.error(`Getting git repository url failed`, error);
+    }
+
+    if (!fullUrl) {
+        return getDefaultRepository();
+    }
+
+    if (fullUrl.endsWith(GIT_REPO_URL_SUFFIX)) {
+        fullUrl = fullUrl.substring(0, fullUrl.length - GIT_REPO_URL_SUFFIX.length);
+    }
+
+    const defaultUrl = getDefaultUrlForType(type);
+    const defaultUrlIndex = fullUrl.indexOf(defaultUrl);
+    if (defaultUrlIndex >= 0) {
+        const name = fullUrl.substring(defaultUrlIndex + defaultUrl.length + 1); // +1 is for the divider after domain
```

Comment:
I guess we can just pass the url to `URL` constructor and get the pathname (except leading `/`), without the need for knowing the type of the URL.

Reply:
url can be not an actual url like git@github.com:zeplin/vscode-extension, so we have to do this manually